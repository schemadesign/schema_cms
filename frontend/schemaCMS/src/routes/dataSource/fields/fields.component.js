import React, { Fragment, PureComponent } from 'react';
import PropTypes from 'prop-types';
import Helmet from 'react-helmet';
import { FormattedMessage } from 'react-intl';

import messages from './fields.messages';
import DataPreview from '../../../shared/components/dataPreview/dataPreview.component';
import { ContextHeader } from '../../../shared/components/contextHeader';
import { DataSourceNavigation } from '../../../shared/components/dataSourceNavigation';
import { MobileMenu } from '../../../shared/components/menu/mobileMenu';
import { filterMenuOptions } from '../../../shared/utils/helpers';
import { getProjectMenuOptions } from '../../project/project.constants';
import { Container } from './fields.styles';

export class Fields extends PureComponent {
  static propTypes = {
    userRole: PropTypes.string.isRequired,
    previewData: PropTypes.object.isRequired,
    fetchPreview: PropTypes.func.isRequired,
    intl: PropTypes.object.isRequired,
    dataSource: PropTypes.object.isRequired,
    history: PropTypes.object.isRequired,
    match: PropTypes.object.isRequired,
  };

  render() {
    const headerTitle = this.props.dataSource.name;
    const headerSubtitle = <FormattedMessage {...messages.subTitle} />;
    const menuOptions = getProjectMenuOptions(this.props.dataSource.project.id);

    return (
      <Container>
        <Helmet title={this.props.intl.formatMessage(messages.pageTitle)} />
        <MobileMenu
          headerTitle={headerTitle}
          headerSubtitle={headerSubtitle}
          options={filterMenuOptions(menuOptions, this.props.userRole)}
        />
        <ContextHeader title={headerTitle} subtitle={headerSubtitle}>
          <DataSourceNavigation {...this.props} />
        </ContextHeader>
        <DataPreview {...this.props} />
        <DataSourceNavigation {...this.props} hideOnDesktop />
      </Container>
    );
  }
}
