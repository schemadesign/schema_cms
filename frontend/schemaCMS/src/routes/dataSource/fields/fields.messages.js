/* eslint-disable max-len */
import { defineMessages } from 'react-intl';

export default defineMessages({
  pageTitle: {
    id: 'dataSource.fields.pageTitle',
    defaultMessage: 'Fields',
  },
  subTitle: {
    id: 'dataSource.fields.subTitle',
    defaultMessage: 'Fields',
  },
});
