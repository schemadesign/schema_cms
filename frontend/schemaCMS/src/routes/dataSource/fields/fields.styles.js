import styled from 'styled-components';

export const Container = styled.div`
  padding-bottom: 50px;
`;
