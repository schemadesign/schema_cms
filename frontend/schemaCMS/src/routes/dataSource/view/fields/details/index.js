export { Details } from './details.component';
