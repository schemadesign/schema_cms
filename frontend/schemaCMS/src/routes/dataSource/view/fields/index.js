export { Fields } from './fields.component';
