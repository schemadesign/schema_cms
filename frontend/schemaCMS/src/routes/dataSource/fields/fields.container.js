import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { withRouter } from 'react-router-dom';
import { hot } from 'react-hot-loader';
import { bindPromiseCreators, promisifyRoutine } from 'redux-saga-routines';
import { compose } from 'ramda';
import { injectIntl } from 'react-intl';

import { Fields } from './fields.component';
import { DataSourceRoutines, selectDataSource, selectPreviewData } from '../../../modules/dataSource';

const mapStateToProps = createStructuredSelector({
  dataSource: selectDataSource,
  previewData: selectPreviewData,
});

export const mapDispatchToProps = dispatch => ({
  ...bindPromiseCreators(
    {
      fetchPreview: promisifyRoutine(DataSourceRoutines.fetchPreview),
    },
    dispatch
  ),
});

export default compose(
  hot(module),
  connect(
    mapStateToProps,
    mapDispatchToProps
  ),
  injectIntl,
  withRouter
)(Fields);
