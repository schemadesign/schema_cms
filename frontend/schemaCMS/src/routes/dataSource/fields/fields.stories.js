import React from 'react';
import { storiesOf } from '@storybook/react';

import { Fields } from './fields.component';
import { withTheme } from '../../../.storybook/decorators';
import { history, intl } from '../../../.storybook/helpers';
import { tableFields as fields, tableData as data } from '../../../shared/utils/dataMock';

export const noDataProps = {
  fetchPreview: Function.prototype,
  previewData: {},
  dataSource: {},
  history,
  intl,
  match: {
    params: {
      step: '2',
    },
  },
};

export const defaultProps = {
  ...noDataProps,
  previewData: { data, fields },
};

storiesOf('Data Source|Fields', module)
  .addDecorator(withTheme())
  .add('no data', () => <Fields {...noDataProps} />)
  .add('default', () => <Fields {...defaultProps} />);
