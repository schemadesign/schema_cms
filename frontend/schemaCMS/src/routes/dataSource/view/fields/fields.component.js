import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import Helmet from 'react-helmet';

import { Container } from './fields.styles';
import messages from './fields.messages';
import { StepNavigation } from '../../../../shared/components/stepNavigation';
import DataPreview from '../../../../shared/components/dataPreview/dataPreview.component';

export class Fields extends PureComponent {
  static propTypes = {
    previewData: PropTypes.object.isRequired,
    fetchPreview: PropTypes.func.isRequired,
    intl: PropTypes.object.isRequired,
    dataSource: PropTypes.object.isRequired,
    history: PropTypes.object.isRequired,
    match: PropTypes.object.isRequired,
  };

  render() {
    return (
      <Container>
        <Helmet title={this.props.intl.formatMessage(messages.pageTitle)} />
        <DataPreview {...this.props} />
        <StepNavigation {...this.props} />
      </Container>
    );
  }
}
