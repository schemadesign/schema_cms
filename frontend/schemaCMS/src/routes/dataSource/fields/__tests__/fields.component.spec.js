import React from 'react';
import { shallow } from 'enzyme';

import { Fields } from '../fields.component';
import { defaultProps } from '../fields.stories';

describe('Fields: Component', () => {
  const component = props => <Fields {...defaultProps} {...props} />;

  const render = (props = {}) => shallow(component(props));

  it('should render correctly', () => {
    const wrapper = render();

    expect(wrapper).toMatchSnapshot();
  });
});
