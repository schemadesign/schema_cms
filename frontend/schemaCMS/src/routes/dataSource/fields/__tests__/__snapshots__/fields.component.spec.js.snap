// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fields: Component should render correctly 1`] = `
<Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    title="Fields"
  />
  <MobileMenu
    headerSubtitle={
      <FormattedMessage
        defaultMessage="Fields"
        id="dataSource.fields.subTitle"
        values={Object {}}
      />
    }
    options={
      Array [
        Object {
          "allowedRoles": Array [
            "admin",
            "editor",
          ],
          "id": "projectsNavBtn",
          "label": <FormattedMessage
            defaultMessage="Projects"
            id="project.view.projectsListPage"
            values={Object {}}
          />,
          "to": "/project/",
          "type": "link_item",
        },
        Object {
          "allowedRoles": Array [
            "admin",
            "editor",
          ],
          "id": "projectDetailsNavBtn",
          "label": <FormattedMessage
            defaultMessage="Project details"
            id="project.view.projectDetails"
            values={Object {}}
          />,
          "to": "/project/1",
          "type": "link_item",
        },
        Object {
          "allowedRoles": Array [
            "admin",
            "editor",
          ],
          "id": "dataSourceNavBtn",
          "label": <FormattedMessage
            defaultMessage="Data Sources"
            id="project.view.projectDataSources"
            values={Object {}}
          />,
          "to": "/project/1/datasource",
          "type": "link_item",
        },
        Object {
          "allowedRoles": Array [
            "admin",
          ],
          "id": "usersNavBtn",
          "label": <FormattedMessage
            defaultMessage="Users"
            id="project.view.usersPage"
            values={Object {}}
          />,
          "to": "/project/1/user",
          "type": "link_item",
        },
        Object {
          "allowedRoles": Array [
            "admin",
            "editor",
          ],
          "id": "folderNavBtn",
          "label": <FormattedMessage
            defaultMessage="Pages"
            id="project.view.projectPages"
            values={Object {}}
          />,
          "to": "/project/1/folder",
          "type": "link_item",
        },
      ]
    }
  />
  <ContextHeader
    subtitle={
      <FormattedMessage
        defaultMessage="Fields"
        id="dataSource.fields.subTitle"
        values={Object {}}
      />
    }
  >
    <DataSourceNavigation
      dataSource={
        Object {
          "project": Object {
            "id": "1",
          },
        }
      }
      fetchPreview={[Function]}
      hideOnDesktop={false}
      history={
        Object {
          "goBack": [Function],
          "push": [Function],
        }
      }
      intl={
        Object {
          "formatMessage": [Function],
        }
      }
      match={
        Object {
          "params": Object {
            "step": "2",
          },
          "url": "url",
        }
      }
      previewData={
        Object {
          "data": Array [
            Object {
              "Birthday": "02/22/1732",
              "Birthplace": "Westmoreland County",
              "First Name": "George",
              "Full Name": "George Washington",
              "Last Name": "Washington",
              "State of Birth": "VA",
              "Url": "some.url.com",
            },
            Object {
              "Birthday": "10/30/1735",
              "Birthplace": "Braintree",
              "First Name": "John",
              "Full Name": "John Adams",
              "Last Name": "Adams",
              "State of Birth": "MA",
              "Url": "some.url.com",
            },
            Object {
              "Birthday": "04/13/1743",
              "Birthplace": "Shadwell",
              "First Name": "Thomas",
              "Full Name": "Thomas Jefferson",
              "Last Name": "Jefferson",
              "State of Birth": "VA",
              "Url": "some.url.com",
            },
            Object {
              "Birthday": "03/16/1751",
              "Birthplace": "Port Conway",
              "First Name": "James",
              "Full Name": "James Madison",
              "Last Name": "Madison",
              "State of Birth": "VA",
              "Url": "some.url.com",
            },
            Object {
              "Birthday": "04/28/1758",
              "Birthplace": "Monroe Hall",
              "First Name": "James",
              "Full Name": "James Monroe",
              "Last Name": "Monroe",
              "State of Birth": "VA",
              "Url": "some.url.com",
            },
          ],
          "fields": Object {
            "Birthday": Object {},
            "Birthplace": Object {},
            "First Name": Object {},
            "Full Name": Object {},
            "Last Name": Object {},
            "State of Birth": Object {},
            "Url": Object {},
          },
        }
      }
      userRole="admin"
    />
  </ContextHeader>
  <DataPreview
    dataSource={
      Object {
        "project": Object {
          "id": "1",
        },
      }
    }
    fetchPreview={[Function]}
    history={
      Object {
        "goBack": [Function],
        "push": [Function],
      }
    }
    intl={
      Object {
        "formatMessage": [Function],
      }
    }
    jobId={null}
    match={
      Object {
        "params": Object {
          "step": "2",
        },
        "url": "url",
      }
    }
    previewData={
      Object {
        "data": Array [
          Object {
            "Birthday": "02/22/1732",
            "Birthplace": "Westmoreland County",
            "First Name": "George",
            "Full Name": "George Washington",
            "Last Name": "Washington",
            "State of Birth": "VA",
            "Url": "some.url.com",
          },
          Object {
            "Birthday": "10/30/1735",
            "Birthplace": "Braintree",
            "First Name": "John",
            "Full Name": "John Adams",
            "Last Name": "Adams",
            "State of Birth": "MA",
            "Url": "some.url.com",
          },
          Object {
            "Birthday": "04/13/1743",
            "Birthplace": "Shadwell",
            "First Name": "Thomas",
            "Full Name": "Thomas Jefferson",
            "Last Name": "Jefferson",
            "State of Birth": "VA",
            "Url": "some.url.com",
          },
          Object {
            "Birthday": "03/16/1751",
            "Birthplace": "Port Conway",
            "First Name": "James",
            "Full Name": "James Madison",
            "Last Name": "Madison",
            "State of Birth": "VA",
            "Url": "some.url.com",
          },
          Object {
            "Birthday": "04/28/1758",
            "Birthplace": "Monroe Hall",
            "First Name": "James",
            "Full Name": "James Monroe",
            "Last Name": "Monroe",
            "State of Birth": "VA",
            "Url": "some.url.com",
          },
        ],
        "fields": Object {
          "Birthday": Object {},
          "Birthplace": Object {},
          "First Name": Object {},
          "Full Name": Object {},
          "Last Name": Object {},
          "State of Birth": Object {},
          "Url": Object {},
        },
      }
    }
    userRole="admin"
  />
  <DataSourceNavigation
    dataSource={
      Object {
        "project": Object {
          "id": "1",
        },
      }
    }
    fetchPreview={[Function]}
    hideOnDesktop={true}
    history={
      Object {
        "goBack": [Function],
        "push": [Function],
      }
    }
    intl={
      Object {
        "formatMessage": [Function],
      }
    }
    match={
      Object {
        "params": Object {
          "step": "2",
        },
        "url": "url",
      }
    }
    previewData={
      Object {
        "data": Array [
          Object {
            "Birthday": "02/22/1732",
            "Birthplace": "Westmoreland County",
            "First Name": "George",
            "Full Name": "George Washington",
            "Last Name": "Washington",
            "State of Birth": "VA",
            "Url": "some.url.com",
          },
          Object {
            "Birthday": "10/30/1735",
            "Birthplace": "Braintree",
            "First Name": "John",
            "Full Name": "John Adams",
            "Last Name": "Adams",
            "State of Birth": "MA",
            "Url": "some.url.com",
          },
          Object {
            "Birthday": "04/13/1743",
            "Birthplace": "Shadwell",
            "First Name": "Thomas",
            "Full Name": "Thomas Jefferson",
            "Last Name": "Jefferson",
            "State of Birth": "VA",
            "Url": "some.url.com",
          },
          Object {
            "Birthday": "03/16/1751",
            "Birthplace": "Port Conway",
            "First Name": "James",
            "Full Name": "James Madison",
            "Last Name": "Madison",
            "State of Birth": "VA",
            "Url": "some.url.com",
          },
          Object {
            "Birthday": "04/28/1758",
            "Birthplace": "Monroe Hall",
            "First Name": "James",
            "Full Name": "James Monroe",
            "Last Name": "Monroe",
            "State of Birth": "VA",
            "Url": "some.url.com",
          },
        ],
        "fields": Object {
          "Birthday": Object {},
          "Birthplace": Object {},
          "First Name": Object {},
          "Full Name": Object {},
          "Last Name": Object {},
          "State of Birth": Object {},
          "Url": Object {},
        },
      }
    }
    userRole="admin"
  />
</Fragment>
`;
